integration/log_rotation_test.go

```diff
@@ -0,0 +1,153 @@
+	c.Assert(err, checker.IsNil)
+	os.Rename(traefikTestAccessLogFile, traefikTestAccessLogFile+".rotated")
+
+	//continue issuing requests
+	_, err = http.Get("http://127.0.0.1:8000/test2")
+	c.Assert(err, checker.IsNil)
+	_, err = http.Get("http://127.0.0.1:8000/test2")
+	c.Assert(err, checker.IsNil)
+
+	// Verify access.log.rotated output as expected
+	rotated, err := os.Open(traefikTestAccessLogFile + ".rotated")
+	c.Assert(err, checker.IsNil)
+	rotatedLog := bufio.NewScanner(rotated)
+	count := 0
+	for rotatedLog.Scan() {
+		line := rotatedLog.Text()
+		c.Log("rl: " + line)
+		if len(line) > 0 {
+			CheckAccessLogFormat(c, line, count)
+		}
+		count++
+	}
+	c.Assert(count, checker.Equals, 1)
+
+	// Verify access.log output as expected
+	file, err := os.Open(traefikTestAccessLogFile)
+	c.Assert(err, checker.IsNil)
+	accessLog := bufio.NewScanner(file)
+	for accessLog.Scan() {
+		line := accessLog.Text()
+		c.Log("al: " + line)
+		if len(line) > 0 {
+			CheckAccessLogFormat(c, line, count)
+		}
+		count++
+	}
+	// combined, we have 3 lines; i.e. 1 in the initial access.log
+	c.Assert(count, checker.Equals, 3)
+
+	verifyEmptyErrorLog(c, traefikTestLogFile)
+}
+
+func (s *LogRotationSuite) TestTraefikLogRotation(c *check.C) {
+	// Start Traefik
+	cmd, _ := s.cmdTraefik(withConfigFile("fixtures/traefik_log_config.toml"))
+	err := cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+	defer os.Remove(traefikTestAccessLogFile)
+	defer os.Remove(traefikTestLogFile)
+
+	// Allow time to startup
+	time.Sleep(1 * time.Second)
+
+	// issue SIGUSR1 signal to server process and move traefik log
+	err = cmd.Process.Signal(syscall.SIGUSR1)
+	c.Assert(err, checker.IsNil)
+	os.Rename(traefikTestLogFile, traefikTestLogFile+".rotated")
+
+	cmd.Process.Signal(syscall.SIGTERM)
+
+	// Allow time for switch to be processed
+	time.Sleep(3 * time.Second)
+
+	// we have at least 6 lines in traefik.log.rotated
+	rotated, err := os.Open(traefikTestLogFile + ".rotated")
+	c.Assert(err, checker.IsNil)
+	rotatedLog := bufio.NewScanner(rotated)
+	count := 0
+	for rotatedLog.Scan() {
+		line := rotatedLog.Text()
+		c.Log("rl " + line)
+		count++
+	}
+	// GreaterOrEqualThan used to ensure test doesn't break
+	// If more log entries are output on startup
+	c.Assert(count, checker.GreaterOrEqualThan, 6)
+
+	//Verify traefik.log output as expected
+	file, err := os.Open(traefikTestLogFile)
+	c.Assert(err, checker.IsNil)
+	traefikLog := bufio.NewScanner(file)
+	for traefikLog.Scan() {
+		line := traefikLog.Text()
+		c.Log("tl " + line)
+		count++
+	}
+
+	// combined, we have at least 13 lines
+	c.Assert(count, checker.GreaterOrEqualThan, 13)
+}
+
+func verifyEmptyErrorLog(c *check.C, name string) {
+	err := try.Do(30*time.Second, func() error {
+		traefikLog, e2 := ioutil.ReadFile(name)
+		if e2 != nil {
+			return e2
+		}
+		if len(traefikLog) > 0 {
+			fmt.Printf("%s\n", string(traefikLog))
+			c.Assert(traefikLog, checker.HasLen, 0)
```

Comment:
While it may work inside the `try.Do` block, ideally the check should happen outside. A closure could be helpful to achieve that.